### PR TITLE
Add smart-proxy part for OCP Advisor Milestone 2 (Clusters View)

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description
Needed for https://issues.redhat.com/browse/CCXDEV-6487

This endpoints returns a list of all clusters retrieved from AMS API and a "summary" for each cluster:
- `last_checked_at` = the time of last clusters analysis
- `total_hit_count`
- `hits_by_total_risk` = a map of total risk values to be matched on frontend with numbers of hitting rules for each severity

Fixes CCXDEV-6487

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)
- Bump-up dependent library (no changes in the code)
- REST API tests

## Testing steps
alongside aggregator

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: content/content.go [hunk 1]
```diff
@@ -216,6 +216,30 @@ func (s *RulesWithContentStorage) GetExternalRuleIDs() []ctypes.RuleID {
...
+// along with a list of unique severities
+func (s *RulesWithContentStorage) GetRuleSeverities() (
+	severityMap map[ctypes.RuleID]int,
+	uniqueSeverities []int,
+) {
+	s.Lock()
+	defer s.Unlock()
+
+	severityMap = make(map[ctypes.RuleID]int)
+	uniqueMap := make(map[int]bool)
```
Comment (tisnik) [inline content/content.go:229]: `map[interface{}]interface{}` is usually used as a set

**Review by tisnik — APPROVED** [review]
LGTM, nice approach

File: content/content.go (hunk 1 above)
Comment (Bee-lee) [reply content/content.go]: thank you, we know (need) the type of the keys, so I changed it to `map[int]interface{}`

File: server/handlers_v2.go [hunk 2]
```diff
@@ -289,6 +289,112 @@ func (server HTTPServer) getRecommendations(writer http.ResponseWriter, request
...
+	if err != nil {
+		log.Err(err).Msg(orgIDTokenError)
+		return
+	}
+	log.Info().Int(orgIDTag, int(orgID)).Str(userIDTag, string(userID)).Msg("getClustersView start")
+
+	// get a list of clusters from AMS API
+	clusterInfoList, clusterNamesMap, err := server.readClustersForOrgID(orgID)
+	if err != nil {
+		log.Error().Err(err).Int(orgIDTag, int(orgID)).Msgf("problem reading cluster list for org")
```
Comment (tisnik) [inline server/handlers_v2.go:308]: Msgf -> Msg

File: server/handlers_v2.go [hunk 3]
```diff
@@ -289,6 +289,112 @@ func (server HTTPServer) getRecommendations(writer http.ResponseWriter, request
...
+
+		return
+	}
+	log.Info().Uint32(orgIDTag, uint32(orgID)).Msgf(
+		"getClustersView getting clusters and impacting recommendations from aggregator took %s", time.Since(tStartImpacting),
+	)
+
+	clusterViewResponse, err := matchClusterInfoRuleSeverity(clusterNamesMap, clusterRecommendationMap)
+	if err != nil {
+		log.Error().Err(err).Msgf("error matching cluster list and rule severities")
```
Comment (tisnik) [inline server/handlers_v2.go:330]: Msgf->Msg

File: server/handlers_v2.go [hunk 4]
```diff
@@ -438,6 +544,71 @@ func (server HTTPServer) getImpactingRecommendations(
...
+	if err != nil {
+		log.Error().Err(err).Msgf("problem unmarshalling cluster list")
+		handleServerError(writer, err)
+		return nil, err
+	}
+
+	// #nosec G107
+	aggregatorResp, err := http.Post(aggregatorURL, JSONContentType, bytes.NewBuffer(jsonMarshalled))
+	if err != nil {
+		log.Error().Err(err).Msgf("problem getting response from aggregator")
```
Comment (tisnik) [inline server/handlers_v2.go:577]: another one :)
(btw it might be caught by goconst checker)

**Review by tisnik — APPROVED** [review]
Very nice one!

File: server/handlers_v2.go (hunk 2 above)
Comment (Bee-lee) [reply server/handlers_v2.go]: thanks! done

File: server/handlers_v2.go (hunk 3 above)
Comment (Bee-lee) [reply server/handlers_v2.go]: done

File: server/handlers_v2.go (hunk 4 above)
Comment (Bee-lee) [reply server/handlers_v2.go:577]: thanks, gotta start paying attention to these ones

**Review by juandspy — APPROVED** [review]

**Pull request merged**